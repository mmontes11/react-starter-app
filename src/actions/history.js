import * as types from "constants/actionTypes/history";

export const setHistory = (squareIndex, symbol, step) => ({ type: types.SET_HISTORY, squareIndex, symbol, step });
