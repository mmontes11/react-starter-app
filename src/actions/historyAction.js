import * as types from "../constants/historyActionTypes";

export const setHistory = (squareIndex, symbol, step) => dispatch => {
  dispatch({ type: types.SET_HISTORY, squareIndex, symbol, step });
};
