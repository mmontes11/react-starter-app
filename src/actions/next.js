import * as types from "../constants/actionTypes/next";

export const nextPlayer = () => (dispatch, getState) => {
  dispatch({ type: types.NEXT_PLAYER, step: getState().step });
};
