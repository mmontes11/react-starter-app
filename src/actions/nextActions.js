import * as types from "../constants/nextActionTypes";

export const nextPlayer = () => (dispatch, getState) => {
  dispatch({ type: types.NEXT_PLAYER, step: getState().step });
};
