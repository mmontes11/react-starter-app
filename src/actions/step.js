import * as types from "../constants/actionTypes/step";

export const stepForward = () => ({ type: types.STEP_FORWARD });

export const jumpToStep = step => ({ type: types.JUMP_TO_STEP, step });
