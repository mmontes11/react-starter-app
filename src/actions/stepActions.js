import * as types from "../constants/stepActionTypes";

export const stepForward = () => ({ type: types.STEP_FORWARD });

export const jumpToStep = step => ({ type: types.JUMP_TO_STEP, step });
