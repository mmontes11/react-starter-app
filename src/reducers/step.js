import { STEP_FORWARD, JUMP_TO_STEP } from "constants/actionTypes/step";

const initialState = 0;

const stepReducer = (state = initialState, action) => {
  switch (action.type) {
    case STEP_FORWARD:
      return state + 1;
    case JUMP_TO_STEP:
      return action.step;
    default:
      return state;
  }
};

export default stepReducer;
