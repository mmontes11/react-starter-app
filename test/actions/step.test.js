import { stepForward, jumpToStep } from "../../src/actions/step";

describe("actions/step", () => {
  describe("stepForward", () => {
    it("dispatches a step forward action", () => {
      const action = stepForward();
      expect(action).toBeAnAction();
      expect(action).toMatchSnapshot();
    });
  });
  describe("jumpToStep", () => {
    it("dispatches a jump to step action", () => {
      const action = jumpToStep(1);
      expect(action).toBeAnAction();
      expect(action).toMatchSnapshot();
    });
  });
});
