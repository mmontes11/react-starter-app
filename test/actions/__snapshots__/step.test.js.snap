// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/step jumpToStep dispatches a jump to step action 1`] = `
Object {
  "step": 1,
  "type": "JUMP_TO_STEP",
}
`;

exports[`actions/step stepForward dispatches a step forward action 1`] = `
Object {
  "type": "STEP_FORWARD",
}
`;
